src/dune_engine/build_system.ml

```diff
@@ -442,7 +442,11 @@ end = struct
       match Dpath.Target_dir.of_target dir with
       | Regular (With_context (_, dir)) | Anonymous_action (With_context (_, dir)) ->
         (Build_config.get ()).execution_parameters ~dir
-      | _ -> Execution_parameters.default
+      | Anonymous_action Root | Regular Root -> Execution_parameters.default
```

Comment:
Could you add a comment explaining the difference in behaviour in the two cases above?

Reply:
Actually, I just forbade the execution parameters for `Anonymous Root` and `Regular Root`. There can be no rules in those directories, so there's no need for execution parameters either.